packages/application-shell-connectors/src/components/application-context/normalizers.ts

```diff
@@ -184,9 +175,15 @@ type NormalizedGroupByType = {
   [key: string]: Maybe<TStoreDataFence>[];
 };
 
+type SupportedDataFenceType = 'store';
+type NormalizedDataFenceRecord = Record<
+  SupportedDataFenceType,
+  NormalizedGroupedByResourceType
+>;
+
 export const normalizeAllAppliedDataFences = (
   project: TFetchProjectQuery['project']
-) => {
+): NormalizedDataFenceRecord | null => {
```

Comment:
Same here? Do we need the return type?

Reply:
hmm, I think I get an issue below if I don't do this.
let me check

Reply:
yepp, we can remove this, but I need L214, otherwise we get this error

```
error TS2322: Type '{ store?: NormalizedGroupedByResourceType; } | null' is not assignable to type 'Record<"store", TApplicationContextGroupedByResourceType> | null'.
  Type '{ store?: NormalizedGroupedByResourceType; }' is not assignable to type 'Record<"store", TApplicationContextGroupedByResourceType>'.
    Property 'store' is optional in type '{ store?: NormalizedGroupedByResourceType; }' but required in type 'Record<"store", TApplicationContextGroupedByResourceType>'.

145   dataFences: normalizeAllAppliedDataFences(project),
      ~~~~~~~~~~

  packages/application-shell-connectors/src/components/application-context/application-context.tsx:111:3
    111   dataFences: TApplicationContextDataFences | null;
          ~~~~~~~~~~
    The expected type comes from property 'dataFences' which is declared here on type 'TApplicationContext<AdditionalEnvironmentProperties>'

packages/application-shell-connectors/src/components/application-context/normalizers.ts:178:6 - error TS6196: 'SupportedDataFenceType' is declared but never used.

178 type SupportedDataFenceType = 'store';
```

Reply:
yepp, we can get rid of this.
I needed L214, otherwise we get this error

```bash
error TS2322: Type '{ store?: NormalizedGroupedByResourceType; } | null' is not assignable to type 'Record<"store", TApplicationContextGroupedByResourceType> | null'.
  Type '{ store?: NormalizedGroupedByResourceType; }' is not assignable to type 'Record<"store", TApplicationContextGroupedByResourceType>'.
    Property 'store' is optional in type '{ store?: NormalizedGroupedByResourceType; }' but required in type 'Record<"store", TApplicationContextGroupedByResourceType>'.

145   dataFences: normalizeAllAppliedDataFences(project),
      ~~~~~~~~~~

  packages/application-shell-connectors/src/components/application-context/application-context.tsx:111:3
    111   dataFences: TApplicationContextDataFences | null;
          ~~~~~~~~~~
    The expected type comes from property 'dataFences' which is declared here on type 'TApplicationContext<AdditionalEnvironmentProperties>'

packages/application-shell-connectors/src/components/application-context/normalizers.ts:178:6 - error TS6196: 'SupportedDataFenceType' is declared but never used.

178 type SupportedDataFenceType = 'store';
```